receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -45,11 +49,29 @@ type Preceiver struct {
 var _ receiver.MetricsReceiver = (*Preceiver)(nil)
 
 // New creates a new prometheus.Receiver reference.
-func New(cfg *Configuration) (*Preceiver, error) {
-	if cfg == nil || cfg.ScrapeConfig == nil {
+func New(v *viper.Viper) (*Preceiver, error) {
+	var cfg Configuration
+
+	// Unmarshal our config values (using viper's mapstructure)
+	err := v.Unmarshal(&cfg)
+	if err != nil {
+		return nil, fmt.Errorf("prometheus receiver failed to parse config: %s", err)
+	}
+
+	// Unmarshal prometheus's config values. Since prometheus uses `yaml` tags, so use `yaml`.
+	promCfgMap := v.Sub("config").AllSettings()
```

Comment:
I believe in the /internal helpers we have something to parse YAML from *viper.Viper

Reply:
This is a little different. The utility is to read in a yaml file (represented as bytes) into a viper struct. Here, we already have a viper struct, but instead are marshalling the config values here into a prometheus library defined struct that is tagged with `yaml` tags. `viper` won't marshal into `yaml` tags (as it understands `mapstructure` tags instead), so here we use yaml.v2 to do the final marshalling.